compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -116,7 +116,7 @@ impl<'local, 'interner> Interpreter<'local, 'interner> {
     /// This will handle internal details like binding generics and error handling.
     /// Note that running code which resulted in previous errors during elaboration
     /// may result in similar errors being issued again by the interpreter.
-    pub(crate) fn call_function(
```

Comment:
I am not sure whether it is necessary to expose these comptime interpreter internals to achieve `--force-comptime`. I was thinking we could just pass along the `--force-compile` flag as a frontend/elaborator option and wrap the `main` function body in a `comptime` block. We can then re-use all our existing comptime interpreter logic and can get rid of `execute_cmd/interpret.rs` that was added in this PR. 

One more nit is that we could technically run the comptime interpreter with `nargo compile` instead of `nargo execute` as it is meant to all be evaluated at compile time.